src/git-command-manager.ts

```diff
@@ -242,6 +241,12 @@ class GitCommandManager {
     this.gitEnv[name] = value
   }
 
+  async shaExists(sha: string): Promise<boolean> {
+    const args = ['rev-parse', '--verify', '--quiet', `${sha}^{object}`]
```

Comment:
rev-parse magic :)

`^{object}` means the object can be any type

Reply:
i love git :)

it's a great command line tool that lets you do what you need to do